actionpack/test/dispatch/request/jsonapi_params_parsing_test.rb

```diff
@@ -0,0 +1,53 @@
+require 'abstract_unit'
+
+class JsonapiParamsParsingTest < ActionDispatch::IntegrationTest
+  class TestController < ActionController::Base
+    class << self
+      attr_accessor :last_request_parameters
+    end
+
+    def parse
+      self.class.last_request_parameters = request.request_parameters
+      head :ok
+    end
+  end
+
+  def teardown
+    TestController.last_request_parameters = nil
+  end
+
+  test "parses json params for application json" do
+    payload = "{ \"data\": { \"type\": \"posts\", \"attributes\": { \"title\":"\
+              "\"Hello\", \"date\": \"today\"} , \"relationships\": { "\
+              "\"author\": { \"data\": { \"id\": \"2\", \"type\": \"users\" }"\
+              " }, \"comments\": { \"data\": [ { \"type\": \"comments\","\
+              "\"id\": \"3\" }, { \"type\": \"comments\", \"id\": \"4\" } ]"\
+              " }, \"journal\": { \"data\": null } } } }"
+    expected = { "_type" => "posts", "title" => "Hello", "date" => "today",
+                 "author_id" => "2", "author_type" => "User",
+                 "comment_ids" => ["3", "4"], "comment_types" => ["Comment", "Comment"],
```

Comment:
If you are referring to `ActiveModel::Serializers#from_json` then no, the format is not totally compatible (the associations are referred by `id` and `type`, rather than being nested – the attributes are in a compatible format). It is, however, compatible with `ActiveRecord#new`.
The format outlined in [https://github.com/rails/rails/blob/master/activemodel/lib/active_model/serializers/json.rb#L137](`ActiveModel::Serializers#from_json`) seems a bit unclear about polymorphic relationships. It may make sense to add a `ActiveModel::Serializers#from_jsonapi` method.